src/runtime/list.cpp

```diff
@@ -664,8 +664,7 @@ extern "C" void listIteratorGCHandler(GCVisitor* v, Box* b) {
     v->visit(it->l);
 }
 
-Box* listNew(BoxedClass* cls, Box* container) {
-    assert(cls->cls == type_cls);
+Box* listNew(BoxedClass* cls) {
```

Comment:
I think we should still have the `args` argument to this even if it doesn't get used.